src/components/app/MenuButtons/index.js

```diff
@@ -94,6 +108,110 @@ class MenuButtonsImpl extends React.PureComponent<Props> {
     }
   }
 
+  _deleteThisProfileOnServer = () => {
+    this.setState({
+      metaInfoPanelState: 'delete-confirmation',
+    });
+  };
+
+  _onProfileDeleted = () => {
+    this.setState({
+      metaInfoPanelState: 'profile-deleted',
+    });
+  };
+
+  _resetMetaInfoState = () => {
+    this.setState({
+      metaInfoPanelState: 'initial',
+    });
+  };
+
+  _renderUploadedProfileActions(
+    currentProfileUploadedInformation: UploadedProfileInformation
+  ) {
+    return (
+      <div className="profileInfoUploadedActions">
+        <div className="profileInfoUploadedDate">
+          <span className="profileInfoUploadedLabel">Uploaded:</span>
+          {_formatDate(currentProfileUploadedInformation.publishedDate)}
+        </div>
+        <div className="profileInfoUploadedActionsButtons">
+          <button
+            type="button"
+            className="photon-button photon-button-default photon-button-micro"
+            onClick={this._deleteThisProfileOnServer}
+            title={
+              currentProfileUploadedInformation.jwtToken === null
+                ? 'This profile cannot be deleted because we lack the authorization information.'
+                : null
+            }
+            disabled={currentProfileUploadedInformation.jwtToken === null}
+          >
+            Delete
+          </button>
+        </div>
+      </div>
+    );
+  }
+
+  _renderMetaInfoPanel() {
+    const { currentProfileUploadedInformation } = this.props;
+    const { metaInfoPanelState } = this.state;
+    switch (metaInfoPanelState) {
+      case 'initial': {
+        return (
+          <>
+            <h2 className="metaInfoSubTitle">Profile Information</h2>
+            {currentProfileUploadedInformation
+              ? this._renderUploadedProfileActions(
+                  currentProfileUploadedInformation
+                )
+              : null}
+            <MetaInfoPanel />
+          </>
+        );
+      }
+
+      case 'delete-confirmation': {
+        if (!currentProfileUploadedInformation) {
+          throw new Error(
+            `We're in the state "delete-confirmation" but there's no stored data for this profile, this should not happen.`
+          );
+        }
+
+        const {
+          name,
+          profileToken,
+          jwtToken,
+        } = currentProfileUploadedInformation;
+
+        if (!jwtToken) {
+          throw new Error(
+            `We're in the state "delete-confirmation" but there's no JWT token for this profile, this should not happen.`
+          );
+        }
+
+        const slicedProfileToken = profileToken.slice(0, 6);
+        const profileName = name ? name : `Profile #${slicedProfileToken}`;
+        return (
+          <ProfileDeletePanel
```

Comment:
Quick note: this `<ProfileDeletePanel>` has the capacity to display `<ProfileDeleteSuccess />`, but we don't have the information about the profileToken and jwtToken once it's deleted.
I think I may refactor this a bit in the future, by making `<ProfileDeletePanel>` simpler, and removing `<ProfileDeleteButton>` that's now used only in the uploaded recordings page (I thought I would be able to reuse it here, but it wasn't the case). But I wanted to keep this PR small enough so that's why I kept all of them for now.

Reply:
Thanks for the explanation. Would you mind also adding an explanation to the source code as a comment? I think it's valuable to put it there too.